chain/chain/src/store.rs

```diff
@@ -278,6 +278,22 @@ pub trait ChainStoreAccess {
             Ok(self.get_block_header(hash)?.height())
         }
     }
+
+    /// Get epoch id of the last block with existing chunk for the given shard id.
+    fn get_epoch_id_of_last_block_with_chunk(
+        &mut self,
+        hash: &CryptoHash,
+        shard_id: ShardId,
+    ) -> Result<EpochId, Error> {
+        let mut candidate_hash = hash.clone();
```

Comment:
Why do we do all the clones here if we ever only want &CryptoHash?

Reply:
When we call `self.get_block_header` in loop, we borrow `*self` mutably, and if we store just a reference to needed hash, we can't unborrow `*self`, which leads to 
```
error[E0499]: cannot borrow `*self` as mutable more than once at a time
```

Reply:
Confirmed with @matklad